### PR TITLE
Add psalm

### PR DESCRIPTION
**Pull request by Vesquen**
| Q             | A
| ------------- | ---
| Branch        | master for features and deprecations
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Documented?   | yes
| Fixed tickets | comma-separated list of tickets fixed by the PR, if any

It adds a task for Psalm

# New Task Checklist:

- [x] Is the README.md file updated?
- [x] Are the dependencies added to the composer.json suggestions?
- [x] Is the doc/tasks.md file updated?
- [x] Are the task parameters documented?
- [x] Is the task registered in the tasks.yml file?
- [x] Does the task contains phpspec tests?
- [x] Is the configuration having logical allowed types?
- [x] Does the task run in the correct context?
- [x] Is the `run()` method readable?
- [x] Is the `run()` method using the configuration correctly?
- [x] Are all CI services returning green?

### CODE REVIEW

File: src/Task/Psalm.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+     */
+    public function getConfigurableOptions()
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults([
+            'config' => null,
+            'ignore_patterns' => [],
+            'no_cache' => false,
+            'report' => null,
+            'threads' => 1,
```
Comment (veewee) [inline src/Task/Psalm.php]: Maybe it is better to set this to `null` and let psalm decide what it uses by default?

File: doc/tasks/psalm.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+```yaml
+# grumphp.yml
+parameters:
+    tasks:
+        psalm:
+            config: psalm.xml
+            ignore_patterns: []
+            no_cache: false
+            report: ~ 
+            threads: 1
```
Comment (veewee) [inline doc/tasks/psalm.md]: The threads options is not described in the documentation

File: src/Task/Psalm.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        if (0 === count($files)) {
+            return TaskResult::createSkipped($this, $context);
+        }
+
+        $arguments = $this->processBuilder->createArgumentsForCommand(self::TASK_NAME);
+        $arguments->addOptionalArgument('--config=%s', $config['config']);
+        $arguments->addOptionalArgument('--report=%s', $config['report']);
+        $arguments->addOptionalArgument('--no-cache', $config['no_cache']);
+        $arguments->addOptionalArgument('--threads=%d', $config['threads']);
+        $arguments->addFiles($files);
```
Comment (veewee) [inline src/Task/Psalm.php]: During `run`, the file list might get very big. Would it make sense to only add the files during pre-commit context and run on all files during run context?

Comment (Vesquen) [reply src/Task/Psalm.php]: That does make sense, will fix!

**Comment by janvernieuwe** [comment]
Thanks for adding the docs, we'll merge it in with the next milestone release!

**Pull request merged**